src/hooks/backgroundTrackingTask.tsx

```diff
@@ -0,0 +1,62 @@
+import { useEffect } from "react";
+
+import * as Location from "expo-location";
+import * as TaskManager from "expo-task-manager";
+import { setRecoil } from "recoil-nexus";
+
+import { displayError } from "@utils/error";
+import { userGeolocationState } from "@atoms/geolocationAtom";
+
+export const LOCATION_TASK_NAME = "LOCATION_BACKGROUND_TRACKING";
+
+// Define the background task for location tracking
+TaskManager.defineTask(
+  LOCATION_TASK_NAME,
+  async ({ data, error }: { data: any; error: any }) => {
+    if (error) {
+      displayError(error);
+      return;
+    }
+    if (data) {
+      const { latitude, longitude } = data.locations[0].coords;
+      // Delaying the update of the geolocation state to avoid getting setRecoil is not a function error
+      setTimeout(() => {
+        setRecoil(userGeolocationState, {
+          latitude,
+          longitude,
+          timestamp: Date.now(),
+        });
+      }, 1000);
```

Comment:
So this is delaying it by 1 second *each time* we get an update for the location. Instead you want to wrap this whole piece of code to delay it until it's loaded, not delaying the update each time.